receiver/prometheusreceiver/internal/starttimemetricadjuster.go

```diff
@@ -15,25 +18,55 @@ var (
 	errNoStartTimeMetrics             = errors.New("start_time metric is missing")
 	errNoDataPointsStartTimeMetric    = errors.New("start time metric with no data points")
 	errUnsupportedTypeStartTimeMetric = errors.New("unsupported data type for start time metric")
+
+	// collectorStartTime is the approximate start time of the collector. Used
+	// as a fallback start time for metrics that don't have a start time set.
+	// Set when the component is initialized.
+	collectorStartTime *time.Time
 )
 
+func init() {
+	p, err := process.NewProcess(int32(os.Getpid()))
+	if err != nil {
+		panic("Unable to find current process" + err.Error())
```

Comment:
I'm a little concerned about being able to panic in code that supports a feature-gated feature. I understand these are very unlikely panics, but is that the right behavior for this error path?

Reply:
After some playing around, and given that this is a featuregated feature - I think its simpler to just approximate the collector start time using time.Now() in an init function. No error paths this way and this is supposed to be an approximation anyway.